tools/utils.py

```diff
@@ -64,11 +64,41 @@ def wait_for_droplet_shutdown(droplet):
                         act.type, act.status))
                     if act.status == 'completed':
                         return
+                    if act.status == 'errored':
+                        power_off_droplet(droplet)
+                        return
             time.sleep(4)
     except Exception as err:
         print('   Exception: {}'.format(err))
         raise
 
+def wait_for_droplet_power_off(droplet):
+    try:
+        actions = droplet.get_actions()
+        while True:
+            for act in actions:
+                if act.type == 'power_off':
+                    act.load()
+                    print('   Action {} is {}'.format(
+                        act.type, act.status))
+                    if act.status == 'completed':
+                        return
+                    if act.status == 'errored':
+                        destroy_droplet_and_exit(droplet)
+                        return
+            time.sleep(4)
+    except Exception as err:
+        print('   Exception: {}'.format(err))
+        raise
+
+def power_off_droplet(droplet):
+    print('Powering down droplet with power_off...')
+    try:
+        droplet.power_off(return_dict=True)
+        wait_for_droplet_power_off(droplet)
+    except Exception as err:
+        print('   Exception: {}'.format(err))
+        destroy_droplet_and_exit(droplet)
```

Comment:
I think you can raise the error here instead of using `destroy_droplet_and_exit(droplet)`

Reply:
```suggestion
        raise
```